### PR TITLE
PR for Outputs: automating-tasks, checking-repository-health, set-up-continuous-integration, setting-up-linters, setting-up-your-repository, solving-the-problem, using-design-patterns, writing-tests, writing-html-js-css, delivering the product, reflecting on the learning-experience 

### PR DESCRIPTION
**Pull request by johnkingzy**
Writing-html-js-css -  Delivers a folder containing the template for the proposed Frontend of your project.

Writing-tests - Delivers a folder that contains my test files and its dependencies.

Using-design-patterns - Fulfills a Pull Request showing two Design Patterns.

Solving-the-problem - Delivers a file containing the implementation.

Setting-up-your-repository - Contains the repository for your proposed project

Setting-up-linters - Delivers two screenshots of ESLint and HoundCi

Setting-up-continuous-integration - Delivers a Link to a Successful Travis Build

Checking-repository-health - Delivers a Screenshot of my code coverage and GPA badges

Automating Task - Delivers a screencast showing gulp watching and refreshing my browser

Delivering my Product - Delivers a release notes for my final products

Reflecting on the learning-experience - Delivers a Medium Post on my learning experience while producing these outputs

### CODE REVIEW

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
+/*eslint-disable */
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: why the need for eslint-disable here and eslint-enable below?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+      const tokens = InvertedIndex.tokenize(`${title} ${text}`);
+      const indicies = this.indicies[fileName];
+      tokens.forEach((token) => {
+        if (token in indicies) {
+          const eachToken = indicies[token];
+          if (eachToken.indexOf(bookIndex) === -1) {
+            indicies[token].push(bookIndex);
+          }
+        } else {
+          // Initially this is what happens
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: not a very intuitive comment. I don't know what this means

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+ * @class InvertedIndex
+ * @classdesc blah blah
+ */
+class InvertedIndex {
+  /**
+   * * @constructor
+   * initialises the class base properties
+   */
+  constructor() {
+    this.indicies = {};
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: correct spelling _indices_

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+      const error = `${fileName} has an Invalid File extension, JSON only`;
+      throw new Error(error);
+    }
+    try {
+      JSON.parse(fileContent);
+    } catch (e) {
+      const error = `OOPS!!! ${fileName} is not well formatted`;
+      throw new Error(error);
+    }
+    const content = JSON.parse(fileContent);
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: you should replace line 133 with this line

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    if (content.length === 0) {
+      const error = `${fileName} is an empty JSON file`;
+      throw new Error(error);
+    }
+    content.forEach((elem) => {
+      if (!Object.keys(elem).includes('title')
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: no need to use Object.keys here, just check !elem.title || !elem.text

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+  /**
+   * @searchIndex method
+   * @param {keyword} keyword
+   * @param {locations} locations
+   * @returns {Boolean}
+   * reads the content of the book
+   */
+  searchIndex(keyword, locations) {
+    const self = this;
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: no need to assign this
you are using arrow functions below, scope is retained

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/css/custom.css [hunk 7]
```diff
@@ -0,0 +1,347 @@
+ .image-container {
+    min-height: 100vh;
+    background-position: center center;
+    background-size: cover;
+    position: relative;
+    }
+
+    body {
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/css/custom.css]: inconsistent indentation and position of curly braces in this file, and move body {} to the top

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+  app.filter('rowFunc', rowFilterFactory);
+  mainCtrl.$inject = ['$scope', '$timeout'];
+ function mainCtrl ($scope, $timeout) {
+  const invertedIndex = new InvertedIndex();
+  $scope.uploadedFiles = {};
+  $scope.indexed = {};
+  $scope.welcome = true;
+  $scope.numOfIndexed;
+  let selectedId = document.getElementById("selectall");
+  if( typeof $scope.selection != 'undefined' || $scope.selection != null ) {
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js:14]: use !== for safe comparison

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+        <h4 class="side">Search</h4>
+        </br>
+        <div class="form-group">
+            <center><label class="sr-only" for="keywords">keyword</label>
+            <input type="text" ng-model="keyword" id="system-search" placeholder="Type Keyword">
+            <button type="submit" class="btn btn-default search-btn" ng-click="searchIndex()">Search</button></center>
+            <div class="search-container">
+            <ul>
+            <center><h3 style="color: #cc0000; font-size: 14px; text-decoration: underline;">Select Files</h3>
+                <li class="checkbox" ng-repeat="eachBooks in numOfIndexed">
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html]: eachBooks, semantics

why not just book

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+		<table class="table table-list-search" >
+                    <thead>
+                        <tr>
+                            <th>Index</th>
+							<th ng-repeat="values in numOfBooks">Book {{values+1}}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr ng-repeat="value in allToken">
+                            <td style="text-transform: capitalize;">{{value}}</td>
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html]: token in tokens, instead of value in allToken

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+  var fileselect = document.getElementById("fileselect");
+  fileselect.addEventListener("change", filesFunc, false);
+}
+
+// call initialization file
+if (window.File && window.FileList && window.FileReader) {
+  Init();
+}
+
+function filesFunc (e) {
```
Comment (emmyakin) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js:135]: function name could be better

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 1 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Ok thanks, i just discovered a more better way to do it using "Singleton in ES6"

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 3 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Ok thanks, i will see to that.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 2 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Ok thanks

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js (hunk 11 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js:135]: Thanks, this has been updated

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js (hunk 8 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/js/app.js:14]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 1 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 2 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 4 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 5 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js (hunk 6 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertindex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/css/custom.css (hunk 7 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/lib/css/custom.css]: Thank this has been updated

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html (hunk 9 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html (hunk 10 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html]: thanks this has been updated

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Link to YOUTUBE VIDEO
+[Here is the Link to the Video](https://www.youtube.com/watch?v=4TKLgEn7cDk&feature=youtu.be "Automating Task using G`ulp and Browser-Sync")
+
+- [![IMAGE ALT TEXT HERE](http://img.youtube.com/vi/4TKLgEn7cDk/0.jpg)](http://www.youtube.com/watch?v=4TKLgEn7cDk)
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/output/README.md:10]: Good video. Your output is to show gulp watching and refreshing your browser when changes are made to your HTML, CSS & **_JS files_** . You seem to be missing the changes made to your _**JS files**_

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+#### ScreenShot for my code coverage
+- This folder contains the screenshots for my code coverage and GPA badges
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:7]: I see your badges. Cool

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md [hunk 14]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+## Link to Release Note on Google Docs
+
+- [Preview on Google Doc](https://docs.google.com/document/d/1Qdm7NfE8XWAS1TVRkXJCbhDsDkmu7yNr9CUBXUxNMg4/edit)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:8]: Please provide access

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md [hunk 15]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: Medium
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Link to mu Medium Post 
+
+- [Medium Post](https://medium.com/@maximufkingzy/the-adventure-experience-so-far-andela-3c3f6ccbcfa1)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:8]: Nice read. One question though - Is HTML/CSS a concept?

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: TravisCI
+- **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
+- **With**: *independently*
+
+#### Link to a Successful Travis Build
+[Link to a Successful travis build](https://travis-ci.org/andela-ksolomon/invertedIndex)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:7]: Nice

File: 06-building-a-program-start-to-finish/setting-up-linters/output/README.md [hunk 17]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Terminal & GitHub
+- **Deliver**: two screenshots of ESLint [Airbnb](https://github.com/airbnb/javascript)  style violations in the terminal and [HoundCI](https://houndci.com/) comment on a PR
+- **With**: *independently*
+
+#### ScreenShot
+The Screenshot of ESLint [Airbnb] style violations in the terminal is my outputs folder
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-linters/output/README.md:7]: Cool

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md [hunk 18]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: the repository for your proposed project and at least one `Pull Request`
+- **With**: *independently*
+
+#### Link To my Github repo
+[my github repo](https://github.com/andela-ksolomon/invertedIndex/tree/developing)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:7]: - Your "How to Use" section is not well descriptive.
- Replace <THIS_REPOSITORY_URL> with your actual Url.
- Your entire readme should be look better than this

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+  /**
+   * @createIndex method
+   * @param {fileName} fileName
+   * @param {fileContent} fileContent
+   * @returns {boolean}
+   * create index of the fileName
+   */
+  createIndex(fileName, fileContent) {```
+
+To use the methods of the object you first have to create an instance of the class by doing: myClass = new InvertedIndex() then myclass.createIndex();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:39]: Ok.

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+
+            $(':file').on('fileselect', function (event, numFiles, label) {
+                var input = $(this).parents('.input-group').find(':text'),
+                    log = numFiles > 1 ? numFiles + ' files were' : label + ' was ';
+                if (input.length) {
+                    input.val(log);
+                }
+            });
+        });
+    });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/index.html:213]: Why do you have this piece of JavaScript here?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 21]
```diff
@@ -0,0 +1,122 @@
+/* My Test Setups */
+/* global describe it expect */
+import myBook from './books.json';
+import emptyBook from './emptyBook.json';
+import invalidKeys from './Invalidkeys.json';
+
+/* My Suites */
+describe('Class and Method Instantaion', () => {
+  const myClass = new InvertedIndex();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js:9]: It is advised that when you create an instance of a class it should exactly match the name.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 22]
```diff
@@ -0,0 +1,122 @@
+/* My Test Setups */
+/* global describe it expect */
+import myBook from './books.json';
+import emptyBook from './emptyBook.json';
+import invalidKeys from './Invalidkeys.json';
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js:6]: A base filename should exactly match the name of its default export. [See link](https://github.com/airbnb/javascript#naming--filename-matches-export)

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 23]
```diff
@@ -0,0 +1,122 @@
...
+    expect(typeof InvertedIndex.tokenize).toBe('function');
+  });
+
+  it('Should contain the validateFile method', () => {
+    expect(typeof InvertedIndex.validateFile).toBe('function');
+  });
+});
+
+describe('Populating Data', () => {
+  const myClass = new InvertedIndex();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js:44]: I see that you created an instance if your invertedIndex class above. Is there a better way of handling this?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+  it('Should contain the tokenize method', () => {
+    expect(typeof InvertedIndex.tokenize).toBe('function');
+  });
+
+  it('Should contain the validateFile method', () => {
+    expect(typeof InvertedIndex.validateFile).toBe('function');
+  });
+});
+
+describe('Populating Data', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js:43]: What exactly is this test suite testing for? Do you have a method called populateData()?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 25]
```diff
@@ -0,0 +1,221 @@
...
+   * @returns {Array}
+   * get the result of the keyword from the indicies
+   */
+  getResult(keyword, fileName) {
+    const searchResult = {},
+      keywords = InvertedIndex.cleanValues(keyword),
+      fileIndex = this.indicies[fileName],
+      currentToken = Object.keys(this.indicies[fileName]);
+    if (keywords.length === 0) {
+      throw new Error('Search for Aplhanumeric values only not special character');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:194]: Line 194 exceeds the maximum line length of 80. (max-len). Please fix this.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 26]
```diff
@@ -0,0 +1,221 @@
...
+      } else {
+        searchResult[elem] = [];
+      }
+    });
+    return searchResult;
+  }
+
+  /**
+   * @cleanValues method
+   * @param {str} str
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:208]: What is _**str**_? Please be descriptive with your naming. Abbreviations are not allowed.

How is _**str**_ a dataType?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 27]
```diff
@@ -0,0 +1,221 @@
+/* global FileReader */
+/* global InvertedIndex */
+/**
+ * @class InvertedIndex
+ * @classdesc blah blah
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:5]: What is this?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 28]
```diff
@@ -0,0 +1,221 @@
...
+ */
+/* eslint-disable */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:7]: Why are you using this `/* eslint-disable */` here?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 29]
```diff
@@ -0,0 +1,221 @@
...
+    this.indexedFiles = {};
+    this.uploadedFiles = {};
+  }
+  /**
+   * @createIndex method
+   * @param {fileName} fileName
+   * @param {fileContent} fileContent
+   * @returns {boolean}
+   * create index of the fileName
+   */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:25]: All your JSDoc for this file are not properly done. Please review all. Ensure that your @param(s) and @return(s) are properly described also. Use the right dataTypes also.

You probably need to research JSDocs to have a better understanding

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 30]
```diff
@@ -0,0 +1,221 @@
...
+  createIndex(fileName, fileContent) {
+    this.indicies[fileName] = this.indicies[fileName] || {};
+    const numOfBooks = fileContent.length;
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:28]: Please be descriptive with your naming. Abbreviations are not allowed. `numOfBooks`

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 31]
```diff
@@ -0,0 +1,221 @@
...
+        tokens = InvertedIndex.tokenize(`${title} ${text}`),
+        indicies = this.indicies[fileName];
+      tokens.forEach((token) => {
+        if (token in indicies) {
+          const eachToken = indicies[token];
+          if (eachToken.indexOf(bookIndex) === -1) {
+            indicies[token].push(bookIndex);
+          }
+        } else {
+          // Initially this is what happens
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:41]: Why is this here? Is this not feedback you received earlier from your peer review?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 32]
```diff
@@ -0,0 +1,221 @@
...
+    return this.indicies[fileName];
+  }
+
+   /**
+   * @getNumofBooks method
+   * @param {fileName} fileName
+   * @returns {Array}
+   * create index of the fileName
+   */
+  getNumOfBooks(fileName) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:67]: Please be descriptive with your naming. Abbreviations are not allowed `getNumOfBooks`

**Review by andela-moe — COMMENTED** [review]
- Please be descriptive with your naming. Abbreviations are not allowed
- All your JSDoc for your invertedIndex file are not properly done. Please review all. Ensure that your @param(s) and @return(s) are properly described also. Use the right dataTypes also.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js (hunk 26 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js:208]: Thanks this has been updated
